### PR TITLE
Fix SearchBar semantics

### LINKED ISSUE #1228: [Bug] Fix SearchView Accessibility to read out Semantic Hint
Once this issue is addressed

https://github.com/dotnet/maui/issues/1227

we'll need to fix how accessibility properties get applied to search view

### PR DESCRIPTION
**Pull request by rachelkang**
### Description of Change

Fix SearchBar semantics
- On Android, by updating the resource ID used
- On iOS, by ensuring the hint and description get set on the nested TextField 

### Issues Fixed

Fixes #1228

### CODE REVIEW

File: src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformInterop.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public static void setContentDescriptionForAutomationId(View view, String descri
 
     public static View getSemanticPlatformElement(View view) {
         if (view instanceof SearchView) {
-            view = view.findViewById(androidx.appcompat.R.id.search_button);
+            view = view.findViewById(androidx.appcompat.R.id.search_src_text);
```
Comment (jonathanpeppers) [inline src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformInterop.java:87]: This seems like `search_src_text` is correct if this is meant to get the `TextView`:

https://github.com/androidx/androidx/blob/3078d9e8a726ae5f7a28d713cf4427e1bfc8e185/appcompat/appcompat/src/main/java/androidx/appcompat/widget/SearchView.java#L292

`search_button` is the button:

https://github.com/androidx/androidx/blob/3078d9e8a726ae5f7a28d713cf4427e1bfc8e185/appcompat/appcompat/src/main/java/androidx/appcompat/widget/SearchView.java#L298

Is this code trying to get the `TextView`?

File: src/Core/src/Platform/iOS/SemanticExtensions.cs [hunk 2]
```diff
@@ -11,8 +11,33 @@ public static void UpdateSemantics(this UIView platformView, IView view)
...
+
+				if (!string.IsNullOrWhiteSpace(desc))
+				{
+					textField.AccessibilityLabel = desc;
+				}
+
+				if (!string.IsNullOrWhiteSpace(hint))
+				{
+					textField.AccessibilityHint = hint;
+				}
```
Comment (PureWeen) [inline src/Core/src/Platform/iOS/SemanticExtensions.cs:34]: I think you can remove all of this code

And then move this code back outside the else

https://github.com/dotnet/maui/pull/7802/files#diff-49ae98b58eed27aaa20630cb71ec8bd192e3b3bf7a18e7fa66704b47c4a9b181R38-R39

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformInterop.java (hunk 1 above)
Comment (rachelkang) [reply src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformInterop.java:87]: yep! We're trying to get the SearchView, and it looks like that TextView is being uesd as the SearchView.
The SearchButton stuff we were using before was wrong / not quite relevant

File: src/Core/src/Platform/iOS/SemanticExtensions.cs (hunk 2 above)
Comment (rachelkang) [reply src/Core/src/Platform/iOS/SemanticExtensions.cs:34]: oops, yeah. removing the redundancy; thanks for the catch!

File: src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformInterop.java (hunk 1 above)
Comment (PureWeen) [reply src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformInterop.java:87]: Yea, this was initially setup before we changed SearchBar over to not just being the iconified version

https://github.com/dotnet/maui/commit/1d928baa35bdb7a8a9d7bfd9779ef977dc376693

**Review by jonathanpeppers — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**